pkg/sharky/shard_only_test.go

```diff
@@ -0,0 +1,158 @@
+package sharky
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestShard(t *testing.T) {
```

Comment:
I dont understand what this test is doing. Can we have more comments? Why do we need to test these separately?

Reply:
this is needed because the sharky package is under tested.
there is a high level acceptance test `sharky_test.go` but it does not cover the functionality related to releasing slots and making them available for write, on a shard level.
this test covers that part, but you are right, I will add comments to clarify its purpose.